docs/metrics/tp-fp-fn-tn.md

```diff
@@ -0,0 +1,194 @@
+Rather than being computed at the sample level (e.g. per image), they're computed at the instance level (i.e. per object)
+for instances that the model is detecting. When given an image with multiple objects, each inference and each ground truth
+is assigned to one group, and the definitions of the terms are slightly altered:
+
+| Metric | | Description |
+| --- | --- | --- |
+| True Positive | TP | Inference matched with a ground truth |
+| False Positive | FP | Inference that is not matched with a ground truth |
+| <nobr>False Negative</nobr> | FN | Ground truth that is not matched with an inference |
+| True Negative | TN | <p>:kolena-warning-sign-16: **Poorly defined for object detection!** :kolena-warning-sign-16:</p><p>In object detection tasks, a true negative is any non-object that isn't detected as an object. This isn't well defined and as such true negative isn't a commonly used metric in object detection.</p><div>Occasionally, for object detection tasks "true negative" is used to refer to any image that does not have any true positive or false positive inferences.</div> |
+
+In an object detection task, checking for detection correctness requires a couple of other metrics (e.g., Intersection
+over Union (IoU) and Geometry Matching).
+
+#### Single-class
+
+Let’s assume that a matching algorithm has already been run on all predictions and that the matched pairs and unmatched
+ground truths and predictions are given. Consider the following variables, adapted from
+[`match_inferences`][kolena.workflow.metrics.match_inferences]:
+
+| Variable | Type | Description |
+| --- | --- | --- |
+| `matched` | <nobr>`List[Tuple[GT, Inf]]`</nobr> | List of **matched** ground truth and prediction bounding box **pairs** (from Geometry Matcher) |
+| `unmatched_gt` | `List[GT]` | List of **unmatched** ground truth bounding boxes (from Geometry Matcher) |
+| <nobr>`unmatched_inf`</nobr> | `List[Inf]` | List of **unmatched** inference bounding boxes (from Geometry Matcher) |
+| `T` | `float` | Threshold used to filter valid prediction bounding boxes based on their confidence scores |
+
+Then these metrics are defined:
+
+```python
+TP = len([inf.score >= T for _, inf in matched])
+FN = len([inf.score <  T for _, inf in matched]) + len(unmatched_gt)
+FP = len([inf.score >= T for inf in unmatched_inf])
+```
+
+??? example "Example: Single-class Object Detection"
+
+    ![Legends](../assets/images/metrics-tpfpfntn-legends.png)
+
+    ![Single-class example](../assets/images/metrics-tpfpfntn-single-class.png)
+
+    This example includes two ground truths and two inferences, and when computed with an IoU threshold of 0.5 and
+    confidence score threshold of 0.5 yields:
+
+    | TP | FP | FN |
+    | --- | --- | --- |
+    | 1 | 1 | 1 |
+
+#### Multiclass
+
+Like classification, multiclass object detection tasks compute TP / FP / FN per class.
+
+??? example "Example: Multiclass Object Detection"
```

Comment:
I love these collapsible example sections.

Are the legends too big?  I could make it smaller so it fits well with the image (or just make it part of each image).
Should we center the table?  IMO, it looks better when it's centered.

Reply:
What do you think of this? 👍  or 👎 ?
<img width="851" alt="metrics-tpfpfntn-single-class" src="https://github.com/kolenaIO/kolena/assets/17771952/b214ff49-6721-4a20-8753-4173fb763521">

Reply:
Yup that makes sense